### PR TITLE
Catch errors and fix insert zero counts

### PR DESCRIPTION
**Pull request by otfrom**

### CODE REVIEW

File: src/witan/send/chart.clj [hunk 1]
```diff
@@ -317,9 +317,12 @@
...
 (defn comparison-chart-and-table [comparison-defs]
-  (-> comparison-defs
-      (assoc :chart (comparison-chart comparison-defs))
-      (assoc :table (comparison-table comparison-defs))))
+  (try
+    (-> comparison-defs
+        (assoc :chart (comparison-chart comparison-defs))
+        (assoc :table (comparison-table comparison-defs)))
+    (catch Exception e
+      (throw (ex-info "Unable to create comparison table and charts" {:comparison-defs comparison-defs} e)))))
```
Comment (seb231) [inline src/witan/send/chart.clj:325]: I suppose throwing `comparison-defs` will show you where there's something missing/nil?

File: src/witan/send/chart.clj [hunk 2]
```diff
@@ -387,9 +390,32 @@
...
 
-(defn insert-zero-counts [census-data counts]
-  (mapcat (fn [cy] (map (fn [ay] (if-let [ay-map (some (fn [m] (when (#{ay} (:academic-year m)) m)) (val cy))]
-                                   ay-map
-                                   {:calendar-year (:calendar-year (-> cy val first)), :academic-year ay, :population 0}))
-                        (into (sorted-set) (map :academic-year census-data))))
-          (group-by :calendar-year counts)))
+(defn insert-zero-counts
+  ;; FIXME: This assumes it is for academic year.
+  ([census-data counts]
```
Comment (seb231) [inline src/witan/send/chart.clj:395]: Do we want to keep the two arity version if it makes some assumptions?

**Review by seb231 — APPROVED** [review]
Looks good, two things to double check

File: src/witan/send/chart.clj (hunk 2 above)
Comment (otfrom) [reply src/witan/send/chart.clj:395]: no we don't, but I didn't want to break anyone's code that was using it. :-)

File: src/witan/send/chart.clj (hunk 1 above)
Comment (otfrom) [reply src/witan/send/chart.clj:325]: yeah, I was able to see that the history and projection vectors were empty so I knew that the data was getting lost somewhere to this point.

File: src/witan/send/chart.clj (hunk 2 above)
Comment (seb231) [reply src/witan/send/chart.clj:395]: good call

File: src/witan/send/chart.clj (hunk 1 above)
Comment (seb231) [reply src/witan/send/chart.clj:325]: 👍

**Pull request merged**